### PR TITLE
Add support for multiple selection questions

### PR DESCRIPTION
**Pull request by vishnuravi**
# Add support for multiple selection questions

## :recycle: Current situation & Problem
Currently, only a single choice can be selected for a multiple choice question. However, many questionnaires have questions that allow users to select multiple options simultaneously. This answer format is supported by the HL7 SDC Implementation Guide using the `check-box` UI control code with the `questionnaire-item-control` extension.

## :bulb: Proposed solution
Updates `ResearchKitOnFHIR` to recognize the `check-box` code when using the `questionnaire-item-control` extension and render a question that allows the user to select multiple choices. All selected choices are included as answers in the resulting `QuestionnaireResponse`.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vishnuravi** [comment]
To do:

- [ ] Update UI test 
- [ ] Verify the QuestionnaireResponse output matches Android FHIR

File: Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift [hunk 1]
```diff
@@ -151,7 +151,11 @@ extension QuestionnaireItem {
             guard !answerOptions.isEmpty else {
                 throw FHIRToResearchKitConversionError.noOptions
             }
-            return ORKTextChoiceAnswerFormat(style: ORKChoiceAnswerStyle.singleChoice, textChoices: answerOptions)
+            var choiceAnswerStyle = ORKChoiceAnswerStyle.singleChoice
+            if itemControl == "multi-select" {
+                choiceAnswerStyle = .multipleChoice
```
Comment (PSchmiedmayer) [inline Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift:156]: 🎉🚀

**Review by PSchmiedmayer — APPROVED** [review]
Thank you @vishnuravi!

Great to heave for multiple projects, thank you for the fast and nice PR 🚀

File: Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift (hunk 1 above)
Comment (vishnuravi) [reply Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift:156]: There is an inaccuracy here, the correct code to use in this case would be `check-box` as noted in the IG https://fhir-ru.github.io/codesystem-questionnaire-item-control.html. This has been updated in [d2abc38](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/64/commits/d2abc38fdf8f8ec741a72e8a70d71a62aea9ce1d).

Comment (vishnuravi) [reply Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift:156]: There is an error here, `multi-select` appears to not be a valid code per the SDC IG. It should be `check-box`, as shown [here](http://hl7.org/fhir/R4/valueset-questionnaire-item-control.html). I've updated the PR in [d2abc38](https://github.com/StanfordBDHG/ResearchKitOnFHIR/pull/64/commits/d2abc38fdf8f8ec741a72e8a70d71a62aea9ce1d)

**Pull request merged**